service-proxy.js

```diff
@@ -950,7 +949,44 @@ function reapSinglePeer(hostPort) {
         self.deletePeerIndex(serviceName, hostPort);
     }
 
-    self.channel.peers.delete(hostPort);
+    // TODO: info log/stat
+
+    peer.drain({
+        reason: 'reaped for expired advertisement',
+        direction: 'both',
```

Comment:
I think we only want to drain 'out'.

There is a race here where we reject incoming connections on a peer that is being reaped but decides to come back.